test/libsolidity/syntaxTests/iceRegressionTests/recursive_struct_memory.sol

```diff
@@ -11,3 +11,4 @@ contract Test {
 }
 // ----
 // DeclarationError: (157-198): Identifier already declared.
+// TypeError: (268-300): Type struct Test.RecursiveStruct memory[1] memory is only valid in storage.
```

Comment:
Does the test still what it is supposed to test?

Reply:
All the tests need to be looked at again. Also, new tests from issues should be added.

Reply:
@chriseth It does see: https://github.com/ethereum/solidity/issues/8279